cabal.project

```diff
@@ -137,163 +131,71 @@ source-repository-package
 allow-older: *
 allow-newer: *
 
-package cardano-wallet-core
-    ghc-options: -ddump-to-file -ddump-hi
 
-package cardano-wallet-core-integration
-    ghc-options: -ddump-to-file -ddump-hi
+-- ---------------------------------------------------------
+-- Disable all tests by default
+
+-- This mirrors what they do in the cardano-node cabal file.
+
+tests: False
+
+test-show-details: direct
+
+-- Then enable specific tests in this repo
+
+package cardano-wallet
+  tests: True
+
+package cardano-wallet-core
+  tests: True
 
 package cardano-wallet-cli
-    ghc-options: -ddump-to-file -ddump-hi
+  tests: True
 
 package cardano-wallet-launcher
-    ghc-options: -ddump-to-file -ddump-hi
+  tests: True
+
+package cardano-wallet-text-class
+  tests: True
 
 package cardano-numeric
-    ghc-options: -ddump-to-file -ddump-hi
+  tests: True
 
-package text-class
-    ghc-options: -ddump-to-file -ddump-hi
+package cardano-wallet-core-integration
+  tests: True
 
 package cardano-wallet-test-utils
-    ghc-options: -ddump-to-file -ddump-hi
-
-package cardano-wallet
-    ghc-options: -ddump-to-file -ddump-hi
+  tests: True
 
 package strict-non-empty-containers
-    ghc-options: -ddump-to-file -ddump-hi
+  tests: True
 
--- Verbatim
-package cardano-crypto
-  tests: False
-  benchmarks: False
-package contra-tracer
-  tests: False
-  benchmarks: False
-package iohk-monitoring
+-- The following is needed because Nix is doing something crazy.
```

Comment:
This block of package options probably isn't necessary here.

Reply:
It was. It fixed the error from `nix-shell nix/cabal-shell.nix --run "cabal v2-configure -frelease --enable-tests --enable-benchmarks"`. The first section wasn't needed though...

Reply:
It seems like cabal ignores the global `tests: False` flag, and then disabling tests on a selection of difficult packages is enough to make it work. This probjem is not likely to be related to nix - because we are running cabal outside of nix-build.